lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -49,6 +81,24 @@ def convert_inline_anchor node
     end
   end
 
+  def generate_section_numbering node
+    level = node.level
+    doc_attrs = node.document.attributes
+    if node.numbered && level <= (doc_attrs['sectnumlevels'] || 3).to_i
+      if level < 2 && node.document.doctype == 'book'
+        if node.sectname == 'chapter'
+          %(#{(signifier = doc_attrs['chapter-signifier']) ? "#{signifier} " : ''}<span class="section-numbering">#{node.sectnum}</span>)
```

Comment:
I think this class name should be shortened from section-numbering to sectnum. This is a class name which is going to appear very frequently, so we need to be mindful of how much extra visual and virtual weight this adds to the page. I'm open to other suggestions as well.

Reply:
It makes me think that we should try to adhere to a naming convention. For now, I'm using kebab case without any abbreviation or shortening.

I'm fine with `sectnum`, it's also consistent with the `sectnums` attribute.

So the rule would be short/abbreviated words in lowercase with no separator (and preferably consistent with existing AsciiDoc attributes)?

Reply:
I'm inclined to say yes to that convention. Though we also have to be mindful of the fact that it won't be perfect, and we may have to make exceptions. But we'll do the best we can to make it intuitive.

Reply:
Should we create a list/table with all the built-in classes/roles that we are using in this new converter? So far:

| Name | Context |
| --- | --- |
| `link` | Section links |
| `anchor` | Section anchors |
| `title` | Paragraph title |
| `sectnum` | Section numbering |
| `subtitle` | Document subtitle |
| `byline` | Author(s) block |
| `author` | Author line |
| `revision` | Revision table |

Reply:
That's a good idea. And it's also forward-looking for the specification. Because I would like the specification to be more clear about what the HTML "API" is so to speak when composing a stylesheet. And this might become part of that.

Reply:
For now, I don't want to do anything fancy, just a plain list. Should I add this (hidden) page in `docs` (i.e., without a navigation)? in https://github.com/asciidoctor/asciidoctor/tree/master/docs/modules/api/pages?

Reply:
Since docs aren't published from this branch, it could really go anywhere for now. We might want to think about point it in a devdocs folder for now, then deciding later if and where it might fit in the user docs.